app/code/Magento/Email/Model/Transport.php

```diff
@@ -100,7 +110,8 @@ public function sendMessage()
 
             $this->laminasTransport->send($laminasMessage);
         } catch (\Exception $e) {
-            throw new MailException(new Phrase($e->getMessage()), $e);
+            $this->logger->error($e);
+            throw new MailException(new Phrase('Unable to send mail. Please try again later.'));
```

Comment:
@engcom-Kilo @zakdma, we have regression here. The original exception should be passed as a 2nd param here to understand why it failed, as it was in the past.

Reply:
FYI here is a PR that fixes this regression https://github.com/magento/magento2/pull/33419